main/src/ca/uwaterloo/flix/language/phase/jvm/JvmBackend.scala

```diff
@@ -39,6 +39,10 @@ object JvmBackend extends Phase[Root, CompilationResult] {
     //
     implicit val r: Root = root
 
+    // TODO: This should be limited to those actually used
+    val erasedRefTypes = List(JvmType.PrimBool, JvmType.PrimChar, JvmType.PrimFloat, JvmType.PrimDouble,
```

Comment:
@magnus-madsen I am a bit uncertain what types to use. I want to have "full" types to work on like `Ref(Bool)` but it feels wrong to use `MonoType`s. JvmType seems a bit superfluous except for the support of void. What do you think about introducing `BackendType` (a bit like the real part of PType/RType from the previous version) that is mostly a copy of `MonoType` except that it is separated into primitive/reference and contains the `JvmName` it is represented by. Void would be missing here but it could be solved by using `Option[BackendType]` whenever relevant. That also seems better then allowing void everywhere as a full type

Reply:
How is it different from JvmType?

Reply:
It actually has Ref/Tuple/Channel etc. JvmType only has primitives and Reference(string) basically

Reply:
And how is it different from MonoType then?

Reply:
It would be split intro primitives and objects, and contain jvmnames for objects

Reply:
And this will eventually replace JvmType?

Reply:
yes

Reply:
OK, makes sense.